routes/apiuserroutes.js

```diff
@@ -59,8 +59,35 @@ router.get('/', async (req,res)=>{
 })
 
 
-router.patch("/", (req,res) => {
-    res.json({"status" : true})
+router.patch("/",async (req,res) => {
+    const {status, error, code}=await validateUser(req.query.appid,"user")
+    if(status==false)
+        res.json({'status' : status, error, code})
+    var q={}
+    console.log(req.body)
+    if(req.query.appid)
+    {
+        q.appid=req.query.appid
+    }
+    var queryUpdate={}
+    const LookUp=[ "firstName", "lastName", "username", "email", "dob", "phoneNumber", "role" ,"password"]
+    for(Item in req.body)
+    {
+        if(LookUp.includes(Item))
+        {  
+            queryUpdate[`${Item}`]=req.body[Item]
+        }
+    }
+    console.log(queryUpdate)
+    console.log(q)
+    try{
+        const updateQuery= await User.updateOne(q, queryUpdate, { runValidators: true, context: 'query'} )
```

Comment:
#### Problem here
``` const updateQuery= await User.updateOne(q, queryUpdate, { runValidators: true, context: 'query'} )```
here ```q= username@password``` in this format but in the database does not have such field as appid
#### FIX
Instead of ```q``` try to use ```username``` as parameter